File: backend/src/main/java/tn/esprit/forum/entities/Condidature.java
package tn.esprit.forum.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.forum.entities.Enum.EtatCondidature;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Condidature {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    Long idCondidature;
    EtatCondidature etatCondidature;
    String coverLetter;

    @ManyToOne(cascade = CascadeType.ALL)
    Entretien entretien;

    @ManyToOne
            @JsonIgnore
    Offre offre;

    @ManyToOne(cascade = CascadeType.ALL)
    User user;
    EtatCondidature etatCond;


}

File: frontend/src/app/core/models/condidature.ts
import { Entretien } from './Entretien';
import { offre } from './offre';
import { User } from './user';

export enum EtatCondidature{
  Waitlisted="Waitlisted",
  Accepted="Accepted",
  Rejected="Rejected",
}


export class Condidature {
    idCondidature: number;
    etatCondidature: EtatCondidature;
    coverLetter: string;
    entretien: Entretien;
    offre: offre;
    user: User;
  }
  
  // export interface Entretien {
  //   // Définissez les propriétés de l'entité Entretien si nécessaire
  // }
  
  // export interface Offre {
  //   // Définissez les propriétés de l'entité Offre si nécessaire
  // }
  

  